### PR TITLE
Replace -1 with SPR_NONE for viewport sprite, sort some includes

### PR DESCRIPTION
**Pull request by Broxzier**

### CODE REVIEW

**Comment by Gymnasiast** [comment]
It seems that SPR_NONE and SPRITE_INDEX_NULL serve the same purpose, although it seems that the latter has been used a lot more often. I'd prefer if you replaced SPR_NONE with SPRITE_INDEX_NULL. (Do not do a full find-and-replace, just replaced the calls you already did in the PR.)

**Comment by rwjuk** [comment]
@Gymnasiast Any reason not to replace all the `SPR_NONE` occurrences if they're redundant? Might as well get them all at once.

**Comment by Broxzier** [comment]
@Gymnasiast: Will do. I've also just found 3 more `-1`s in the code that refer to sprites. I also agree with @rwjuk, having a duplicated define sounds unnecessary.

**Comment by Gymnasiast** [comment]
@rwjuk There is a difference in how they're defined:
```
SPR_NONE = -1
SPRITE_INDEX_NULL = 0xFFFF
```
Replacing all SPR_NONE occurrences will also require their receivers to be changed to uint16.

**Comment by rwjuk** [comment]
@Gymnasiast Ahhhh, didn't realise that. Yes, in that case, don't change them all (without changing the receivers, and that should probably be a separate PR anyway)

**Comment by Gymnasiast** [comment]
@rwjuk That's indeed exactly what I intended. I want to keep PRs manageable.

**Comment by rwjuk** [comment]
Yeah, I should've checked that they were absolutely identical first.

**Comment by Broxzier** [comment]
`rct_window::viewport_target_sprite` is apparently the only occurrence where the type does not match. Before this PR there are only 9 occurances of `SPR_NONE`, and to make use of `SPRITE_INDEX_NULL`, `viewport_target_sprite`'s type needs to change.

**Comment by Gymnasiast** [comment]
There is also stuff in testpaint where the receiver is sint32.

**Comment by Broxzier** [comment]
TIL: Don't use \` inside commits. The above commit should say "`rct_window::viewport_target_sprite` needed to change its type to match `SPRITE_INDEX_NULL`".

File: src/openrct2/interface/viewport.c [hunk 1]
```diff
@@ -238,8 +237,8 @@ void viewport_adjust_for_map_height(sint16* x, sint16* y, sint16 *z)
         // of the map. This can happen when the height is larger than the map size.
         sint16 max = gMapSizeMinus2;
         if (pos.x > max && pos.y > max) {
-            sint32 x_corr[] = { -1, 1, 1, -1 };
-            sint32 y_corr[] = { -1, -1, 1, 1 };
+            static const sint32 x_corr[] = { -1, 1, 1, -1 };
```
Comment (Gymnasiast) [inline src/openrct2/interface/viewport.c]: Why this change? It appears unrelated.

**Comment by marijnvdwerf** [comment]
Aren't you guys mixing up `sprites` and `SPR_` constants?

File: src/openrct2/interface/viewport.c (hunk 1 above)
Comment (Broxzier) [inline src/openrct2/interface/viewport.c]: Whoops that slipped in. I'll revert it.

**Comment by Broxzier** [comment]
@marijnvdwerf What's the difference? `SPR_NONE` is used only at 5 places, and seems to be used for sprite indices too.

**Comment by IntelOrca** [comment]
@Broxzier SPR_NONE is for G1 and G2 graphics. SPRITE_INDEX_NULL is for the moving things in the game, e.g. peeps, vehicles, ducks, particles and also litter.
